Adds PR labels definitions to contribution guide (#6413)
jenkinsci/jenkins: docs/MAINTAINERS.adoc

Many features would be better introduced as plugins that have separate release cycles and allow delivering changes faster.
We want to extend the Jenkins core and incorporate widely used functionality and extension points there,
but we try to keep the core as minimal as possible in terms of dependencies.

> When the motivation of the changes is not clear, the pull requests needs to be labeled with `needs-justification`.

Reviewer: ```suggestion
When the motivation of the pull request is unclear, incomplete, or not entirely cogent, the pull request needs to be labeled with `needs-justification`.
```

  We have a semi-automated process that is based on pull request summaries and labels.
  Core maintainers are expected to validate the entries as a part of the pull request review/merge process.
  See the checklist below in the _Merge process_ section.

> * When the proposed changes requires documentation but it is missing, use `needs-docs` label on the pull-request.

Reviewer: ```suggestion
* When the proposed change lacks sufficient documentation, use the `needs-docs` label on the pull request.
```

And we are fine with that, especially for small patches.
Pull requests often become a venue to discuss feasibility, underlying technical decisions and design.
We are fine with that as well.

> The pull requests can be marked with `needs-more-review` to get more eyes on the changes.

Reviewer: ```suggestion
The pull request can be marked with `needs-more-review` to get more eyes on the change.
```

  Not everything is going to be merged, and reviewers are expected to focus on the Jenkins ecosystem integrity first.
  We guide contributors and help them to get their changes integrated, but it needs cooperation on both sides.
  It is **fine** to close invalid and inactive pull requests if there is no activity by a submitter or other contributors.

>   When a pull-request has no activities in the last 30days, it can be marked with the label `stalled`.
>   If there is no consensus or activities after 30 more days, the pull-request can be marked as `proposed-for-close`.
>   The pull-request will then be closed in the next 96hrs with no new activities.

Reviewer: ```suggestion
  When a pull request remains inactive for a month, it can be marked with the label `stalled`.
  If the pull request remains inactive or without consensus for yet another month, the pull request can be marked as `proposed-for-close`.
  The pull request should then be closed in approximately a week if this state persists.
```
